File: BAEKJOON/java/17608_막대기.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static int n, h, answer = 0;
    public static List<Integer> list = new ArrayList<>();

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        input();
        solution();

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            h = Integer.parseInt(br.readLine());
            list.add(h);
        }
    }

    private static void solution() throws IOException {
       int max_num = -1;

       for (int i = list.size() - 1; i >= 0; i--) {
           if (list.get(i) > max_num) {
               max_num = Math.max(max_num, list.get(i));
               answer++;
           }
       }

       bw.write(String.valueOf(answer));
    }
}

File: BAEKJOON/cpp/17608_막대기.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int n, h, answer = 0;
vector<int> v;

void input() {
	cin >> n;

	for (int i = 0; i < n; i++) {
		cin >> h;
		v.push_back(h);
	}
}

void solution() {
	int max_num = -1;

	for (int i = v.size() - 1; i >= 0; i--) {
		if (v[i] > max_num) {
			max_num = max(max_num, v[i]);
			answer++;
		}
	}

	cout << answer;
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	input();
	solution();

	return 0;
}